src/visuals.js

```diff
@@ -1226,6 +1269,7 @@ export const Visuals = {
       body.witherSteps++
       if (body.witherSteps > WITHERING_STEPS) {
         this.witheringBodies = this.witheringBodies.filter((b) => b !== body)
+        p.pop()
```

Comment:
why is this here?

Reply:
this is a bug I found that can interfere with game over text and other stuff. `p.push` then translate is called at the top of the loop and the next line here is `continue`, so it misses the `p.pop` at the end of the loop body and ends up keeping the translation for subsequent drawing.